### PR TITLE
Epoxy 2.0

### PR DESCRIPTION
**Pull request by elihart**
This shows the progress of the Epoxy 2.0 update. I hope to have it done around mid march.

Read more about the goals here: https://github.com/airbnb/epoxy/pull/132

### CODE REVIEW

**Comment by geralt-encore** [comment]
@elihart how close is 2.0 to release? Do you have some ETA?

**Comment by elihart** [comment]
@geralt-encore It's basically feature complete, and I'm itching to release it! The main thing I'm trying to finish right now is more run time validation that models are used correctly (not added twice, not changed after being added). Basically I want to make using it as foolproof as possible, and very clear about when it is misused.

I could push a release candidate to maven if you are interested in starting to use it, it should be stable. I've been integrating this branch into our app and updating some of our more complicated adapters to test that it works well and that I didn't miss any features or have a fundamental design flaw (it's why I realized I needed the Interceptor that was added recently).

Once I finish the validation (hopefully today!) the release candidate should be basically complete. The only other things I was hoping to include are sticky header support and rearranging support which coworkers are working on, but I won't hold the release if those aren't ready.

So basically this should be about ready to use, however, before a public announcement I want to update the sample app and the README to reflect all the changes, since this is quite a big update. I imagine that could take at least another week.

I would love some help updating the sample app, and can give guidance if anybody wants to help with that.

Let me know if you, or anybody else, is interested in a release candidate on maven. I'm planning to do it some time in the next few days anyway. It'd be great to have other people testing and giving feedback!

**Comment by geralt-encore** [comment]
Good to know! I am not sure if I have enough time for updating to 2.0 so no rush for release from my side. I can help you with updating the sample app tho.

**Comment by elihart** [comment]
Validations added with https://github.com/airbnb/epoxy/pull/161. Took me longer than I expected, but it is the last main change I am planning on making.

I will push a release candidate to maven soon, and then start integrating in the airbnb app to test. In parallel I'll also start working on updating the documentation and the sample app.

@geralt-encore It would be great if you could help with the sample app! I can make a list of what needs to be done

**Comment by geralt-encore** [comment]
@elihart sure! List would be perfect.

**Comment by elihart** [comment]
A `2.0.0_RC1` release candidate is now available on Maven. Feel free to try it out, it is stable. I'm going to vet it in our app while we work on the documentation and sample before releasing officially.

**Comment by elihart** [comment]
`2.0.0_RC2` is now released. I improved how we handle hash validation and how options for the annotation processor are set.

**Comment by markus2610** [comment]
2.0.0-RC2

**Comment by elihart** [comment]
@markus2610 whoops... thanks!

**Comment by elihart** [comment]
Also I realized that hiding models in legacy EpoxyAdapters is broken in RC2. I'll be fixing that today

**Comment by elihart** [comment]
Hidden models are fixed with https://github.com/airbnb/epoxy/pull/167 and I'm pushing a `2.0.0-RC3` now

**Comment by elihart** [comment]
I am pushing a `2.0.0-RC4` release now. Hopefully this is the last release candidate, I think we have addressed all the issues so far, but continue to test it in our app.

Work also continues on the new wiki and sample app, which I hope to finish by the end of the week.

**Comment by elihart** [comment]
Just pushed `2.0.0-RC5` which includes some improvements to model groups and better documentation.

The sample is almost done, just going to polish it and record a new gif for the wiki. I'm finishing up the wiki documentation this weekend and will hopefully release everything on Monday

File: README.md [hunk 1]
```diff
@@ -33,39 +28,10 @@ dependencies {
 ```
 
 ## Documentation
-Browse complete documentation at the [Epoxy Wiki](https://github.com/airbnb/epoxy/wiki)
+Seem examples and browse complete documentation at the [Epoxy Wiki](https://github.com/airbnb/epoxy/wiki)
```
Comment (PatrickDattilio) [inline README.md]: Seem -> See

**Comment by elihart** [comment]
@PatrickDattilio Fixed. Thanks!

**Comment by elihart** [comment]
Done :)
